### ISSUE TITLE
Error while backup favorites

### ISSUE THREAD

**Issue by BigDi** [ephes/kptncook#39]
Get a traceback while backuping my 1086 favorites.

╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ C:\Python311\Lib\site-packages\kptncook\__init__.py:167 in backup_kptncook_favorites             │
│                                                                                                  │
│   164 │   │   sys.exit(1)                                                                        │
│   165 │                                                                                          │
│   166 │   fs_repo = RecipeRepository(settings.root)                                              │
│ ❱ 167 │   fs_repo.add_list(recipes)                                                              │
│   168 │   rprint(f"Added {len(recipes)} recipes to local repository")                            │
│   169                                                                                            │
│   170                                                                                            │
│                                                                       
UnicodeEncodeError: 'charmap' codec can't encode character '\u202f' in position 829203: character maps to <undefined>

**Comment by Kadz93** [comment]
super quick fix, the code misses some basic python good practices, I will open a PR

**Comment by ephes** [comment]
Well, pydantic deliberately returns a unicode string when `models.model_dump_json()` is called. And calling write on a file opened in text mode should do the right thing on your operating system. Maybe your default encoding is something like `cp1252`?

Hmm, maybe it's enough just to overwrite the operating system default by calling:
```python
        with self.path.open("w", encoding="utf-8") as f:
            f.write(models.model_dump_json())
```

This is still normal text mode, but would overwrite the default encoding and use utf-8. Can you try this and tell me if it works on windows?

**Comment by Kadz93** [comment]
I can confirm that the additional encoding solves the problem on windows.

**Comment by ephes** [comment]
Cool! I'll [released 0.0.21](https://github.com/ephes/kptncook/releases/tag/0.0.21) containing the fix.
